paper/src/main/java/rocks/gravili/notquests/paper/conversation/ConversationManager.java

```diff
@@ -546,7 +545,7 @@ public void stopConversation(final ConversationPlayer conversationPlayer) {
                 "Skipping stopping conversation, as the conversation you tried to stop is already stopped and some other conversation is running instead.");
         return;
       }
-      if (conversationPlayer.getNpc() != null && conversationPlayer.getNpc() instanceof CitizensNPC) {
+      if (conversationPlayer.getNpc() != null && conversationPlayer.getNpc().isFromCitizens()) {
```

Comment:
mh even like that I would change it. So basically, I try to keep all integration-specific _logic_ (I think that makes it more clear) separated. Meaning everything SPECIFIC to the citizens plugin would be in CitizensNPC or CitizensManager or something like that. So no Citizens-specific logic inside of ConversationManager.

It needs to be as generic as possible and extracted (e.g. a method in NqNPC which is implemented in CitizensNPC and called here). That way, say we decide to add support for zNPCs or something like that in the future, we can implement the same functionality into zNPCs way easier.

Reply:
In fact, the issue is from the name; if the method is renamed "canMove", it's fine ?